metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1349,6 +1419,15 @@ private void replay(ApiMessage message, Optional<OffsetAndEpoch> snapshotId, lon
             case NO_OP_RECORD:
                 // NoOpRecord is an empty record and doesn't need to be replayed
                 break;
+            case BEGIN_TRANSACTION_RECORD:
+                beginTransaction(offset, epoch);
```

Comment:
This is not the real offset when the record if from a snapshot, is there any problem for this?